mlflow/recipes/steps/train.py

```diff
@@ -327,40 +335,69 @@ def my_warn(*args, **kwargs):
-                if self.predict_proba:
-                    model_schema = infer_signature(
-                        raw_X_train, model.predict_proba(raw_X_train.copy())
-                    )
-                    model_info = mlflow.sklearn.log_model(
-                        model,
-                        f"{self.name}/model",
-                        signature=model_schema,
-                        code_paths=self.code_paths,
-                        pyfunc_predict_fn="predict_proba",
-                    )
-                else:
-                    model_schema = infer_signature(raw_X_train, model.predict(raw_X_train.copy()))
-                    model_info = mlflow.sklearn.log_model(
-                        model,
-                        f"{self.name}/model",
-                        signature=model_schema,
-                        code_paths=self.code_paths,
-                    )
-                output_model_path = get_step_output_path(
+
+                trained_pipeline = make_pipeline(transformer, estimator)
+                # Creating a wrapped recipe model for the pipeline so users can achieve both
+                # predict and predict_proba(for a classification problem) correctly.
+                wrapped_model = WrappedRecipeModel(
+                    self.predict_scores_for_all_classes, self.predict_prefix
+                )
+
+                model_uri = get_step_output_path(
                     recipe_root_path=self.recipe_root,
                     step_name=self.name,
                     relative_path=TrainStep.MODEL_ARTIFACT_RELATIVE_PATH,
                 )
-                if os.path.exists(output_model_path) and os.path.isdir(output_model_path):
-                    shutil.rmtree(output_model_path)
-
-                if self.predict_proba:
-                    mlflow.sklearn.save_model(
-                        model, output_model_path, pyfunc_predict_fn="predict_proba"
+                sklearn_model_uri = get_step_output_path(
+                    recipe_root_path=self.recipe_root,
+                    step_name=self.name,
+                    relative_path=TrainStep.SKLEARN_MODEL_ARTIFACT_RELATIVE_PATH,
+                )
+                if os.path.exists(model_uri) and os.path.isdir(model_uri):
+                    shutil.rmtree(model_uri)
+                if os.path.exists(sklearn_model_uri) and os.path.isdir(sklearn_model_uri):
+                    shutil.rmtree(sklearn_model_uri)
+
+                # Saving the sklearn model along with its pyfunc model since mlflow evaluate
+                # predicts results using an sklearn model
+                mlflow.sklearn.save_model(trained_pipeline, sklearn_model_uri)
+                artifacts = {"model_path": sklearn_model_uri}
+                with TempDir() as tmp:
+                    # Saving a temp model so that signature of the model output could be inferred
+                    # before the pyfunc model can be saved as the specified model_uri location
```

Comment:
```suggestion
                    # Saving a temp model so that the output schema (signature) of the model's pyfunc
                    # representation can be inferred and included when logging the model to MLflow Tracking.
                    # Unfortunately, there is currently no easy way to infer the model's signature without first saving
                    # a copy of it, and there is no easy way to add an inferred signature to an existing model
```